crates/ruff_linter/src/rules/pycodestyle/rules/logical_lines/blank_lines.rs

```diff
@@ -0,0 +1,620 @@
+use ruff_diagnostics::AlwaysFixableViolation;
+use ruff_diagnostics::Diagnostic;
+use ruff_diagnostics::Edit;
+use ruff_diagnostics::Fix;
+
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_codegen::Stylist;
+use ruff_python_parser::TokenKind;
+use ruff_source_file::Locator;
+use ruff_text_size::TextSize;
+
+use crate::checkers::logical_lines::LogicalLinesContext;
+
+use super::LogicalLine;
+
+/// Contains variables used for the linting of blank lines.
+#[derive(Debug)]
+#[allow(clippy::struct_excessive_bools)]
+pub(crate) struct BlankLinesTrackingVars {
+    follows_decorator: bool,
+    follows_def: bool,
+    follows_docstring: bool,
```

Comment:
I think we can create an enum instead of using 3 boolean because it is only ever possible that one of those is true, but not all of them at once:

```rust
#[derive(Copy, Clone, Debug, Default)]
enum Follows {
    #[default]
    Other,
    Decorator,
    Def,
    Docstring
}
```